inception-project/inception
# #1567 - Cache knowledge base query results

**What's in the PR**
- Introduce a cache at the level of the KnowledgeBaseService which caches queries on a per-KB basis
- Invalidate caches if the configuration of the KBs of a project is changed
- Use cache for read-only KBs for most queries
- Introduce settings to configure the KB cache behavior

**How to test manually**
* Use entity linking
* Use the KB page
* To see cache usage, set the log level for the KB service to `TRACE`

**Automatic testing**

**Documentation**
